src/Categories/NaturalTransformation/NaturalIsomorphism/Displayed.agda

```diff
@@ -0,0 +1,159 @@
+{-# OPTIONS --safe --without-K #-}
+
+module Categories.NaturalTransformation.NaturalIsomorphism.Displayed where
+
+open import Level
+open import Relation.Binary.Displayed
+
+open import Categories.Category
+open import Categories.Category.Displayed
+open import Categories.Functor
+open import Categories.Functor.Displayed renaming (id′ to idF′)
+open import Categories.Functor.Displayed.Properties
+import Categories.Morphism.Displayed as DMorphism
+import Categories.Morphism.Displayed.Properties as DMorphismₚ
+import Categories.Morphism.Displayed.Reasoning as DMR
+open import Categories.NaturalTransformation.Displayed as DNT hiding (id′)
+open import Categories.NaturalTransformation.NaturalIsomorphism
+
+private
+  variable
+    bo bℓ be bo′ bℓ′ be′ bo″ bℓ″ be″ o ℓ e o′ ℓ′ e′ o″ ℓ″ e″ o‴ ℓ‴ e‴ : Level
+    B C D E : Category o ℓ e
+    F G H I : Functor C D
+    C′ : Displayed C o ℓ e
+    D′ : Displayed D o ℓ e
+    E′ : Displayed E o ℓ e
+
+record DisplayedNaturalIsomorphism
+  {C′ : Displayed C o ℓ e} {D′ : Displayed D o′ ℓ′ e′}
+  (F′ : DisplayedFunctor C′ D′ F) (G′ : DisplayedFunctor C′ D′ G)
+  (F≃G : NaturalIsomorphism F G)
+  : Set (Category.o-level C ⊔ Category.ℓ-level C ⊔ o ⊔ ℓ ⊔ ℓ′ ⊔ e′) where
+  open Displayed C′
+  open NaturalIsomorphism F≃G
+  field
+    F′⇒G′ : DisplayedNaturalTransformation F′ G′ F⇒G
+    F′⇐G′ : DisplayedNaturalTransformation G′ F′ F⇐G
+
+  module ⇒′ = DisplayedNaturalTransformation F′⇒G′
+  module ⇐′ = DisplayedNaturalTransformation F′⇐G′
+
+  field
+    iso′ : ∀ {X} (X′ : Obj[ X ])  → DMorphism.DisplayedIso D′ (⇒′.η′ X′) (⇐′.η′ X′) (iso X)
+
+  module iso′ {X} (X′ : Obj[ X ]) = DMorphism.DisplayedIso (iso′ X′)
+
+open DisplayedNaturalIsomorphism
+
+infix 4 _≃[_]_
+
+_≃[_]_ : ∀ (F′ : DisplayedFunctor C′ D′ F) (F≃G : NaturalIsomorphism F G) (G′ : DisplayedFunctor C′ D′ G) → Set _
+F′ ≃[ F≃G ] G′ = DisplayedNaturalIsomorphism F′ G′ F≃G
+
+_ⓘᵥ′_ : ∀ {F′ : DisplayedFunctor C′ D′ F} {G′ : DisplayedFunctor C′ D′ G} {H′ : DisplayedFunctor C′ D′ H} {G≃H} {F≃G}
+        → G′ ≃[ G≃H ] H′ → F′ ≃[ F≃G ] G′ → F′ ≃[ G≃H ⓘᵥ F≃G ] H′
+_ⓘᵥ′_ {D′ = D′} G′≃H′ F′≃G′ = record
+  { F′⇒G′ = F′⇒G′ G′≃H′ ∘′ᵥ F′⇒G′ F′≃G′
+  ; F′⇐G′ = F′⇐G′ F′≃G′ ∘′ᵥ F′⇐G′ G′≃H′
+  ; iso′ = λ X → DisplayedIso-∘′ (iso′ F′≃G′ X) (iso′ G′≃H′ X)
+  }
+  where
+    open DMorphismₚ D′
+
+_ⓘₕ′_ : ∀ {C : Category bo bℓ be} {D : Category bo′ bℓ′ be′} {E : Category bo″ bℓ″ be″}
```

Comment:
Agda struggled to use the variables declared in the `variable` block here, causing me to add the category, functor, and displayed category arguments explicitly. Not sure why.